# Análise dos Dados do COVID-19

Análise dos Dados do COVID-19 no período entre 22 de Janeiro de 2020 à 27 de Julho de 2020

### Importando Biblíotecas

In [2]:
# Importando o pandas com o apelido pd
import pandas as pd

# Importando os arquivos para a análise
dados_covid = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Analise de Dados/Análise dos Dados do COVID-19/covid_19_clean_complete.csv', sep=',')

In [3]:
# Exibe os dados com 10 primeiras linhas
dados_covid.head(10)

Province/State       Country/Region       Lat        Long  \
0                           NaN          Afghanistan  33.93911   67.709953   
1                           NaN              Albania  41.15330   20.168300   
2                           NaN              Algeria  28.03390    1.659600   
3                           NaN              Andorra  42.50630    1.521800   
4                           NaN               Angola -11.20270   17.873900   
5                           NaN  Antigua and Barbuda  17.06080  -61.796400   
6                           NaN            Argentina -38.41610  -63.616700   
7                           NaN              Armenia  40.06910   45.038200   
8  Australian Capital Territory            Australia -35.47350  149.012400   
9               New South Wales            Australia -33.86880  151.209300   

         Date  Confirmed  Deaths  Recovered  Active             WHO Region  
0  2020-01-22          0       0          0       0  Eastern Mediterranean  
1  2020-01-22          0       0          0       0                 Europe  
2  2020-01-22          0       0          0       0                 Africa  
3  2020-01-22          0       0          0       0                 Europe  
4  2020-01-22          0       0          0       0                 Africa  
5  2020-01-22          0       0          0       0               Americas  
6  2020-01-22          0       0          0       0               Americas  
7  2020-01-22          0       0          0       0                 Europe  
8  2020-01-22          0       0          0       0        Western Pacific  
9  2020-01-22          0       0          0       0        Western Pacific

In [4]:
# Exibe os tipos de cada Dado
dados_covid.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49068 entries, 0 to 49067
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Province/State  14664 non-null  object 
 1   Country/Region  49068 non-null  object 
 2   Lat             49068 non-null  float64
 3   Long            49068 non-null  float64
 4   Date            49068 non-null  object 
 5   Confirmed       49068 non-null  int64  
 6   Deaths          49068 non-null  int64  
 7   Recovered       49068 non-null  int64  
 8   Active          49068 non-null  int64  
 9   WHO Region      49068 non-null  object 
dtypes: float64(2), int64(4), object(4)
memory usage: 3.7+ MB


In [5]:
# Exibe as dimensões da base
dados_covid.shape

(49068, 10)

## Limpando e Identificando Dados Nulos

In [6]:
# Renomeia o índice do dataframe
dados_covid.index.name = 'ID'

# Exibe os Dados
dados_covid.head(10)

Province/State       Country/Region       Lat        Long  \
ID                                                                            
0                            NaN          Afghanistan  33.93911   67.709953   
1                            NaN              Albania  41.15330   20.168300   
2                            NaN              Algeria  28.03390    1.659600   
3                            NaN              Andorra  42.50630    1.521800   
4                            NaN               Angola -11.20270   17.873900   
5                            NaN  Antigua and Barbuda  17.06080  -61.796400   
6                            NaN            Argentina -38.41610  -63.616700   
7                            NaN              Armenia  40.06910   45.038200   
8   Australian Capital Territory            Australia -35.47350  149.012400   
9                New South Wales            Australia -33.86880  151.209300   

          Date  Confirmed  Deaths  Recovered  Active             WHO Region  
ID                                                                           
0   2020-01-22          0       0          0       0  Eastern Mediterranean  
1   2020-01-22          0       0          0       0                 Europe  
2   2020-01-22          0       0          0       0                 Africa  
3   2020-01-22          0       0          0       0                 Europe  
4   2020-01-22          0       0          0       0                 Africa  
5   2020-01-22          0       0          0       0               Americas  
6   2020-01-22          0       0          0       0               Americas  
7   2020-01-22          0       0          0       0                 Europe  
8   2020-01-22          0       0          0       0        Western Pacific  
9   2020-01-22          0       0          0       0        Western Pacific

In [7]:
# Muda o nome da coluna WHO Region para Region
dados_covid.rename(columns={'WHO Region':'Region', 'Country/Region':'Country'}, inplace=True)

# Exibe os dados
dados_covid.head(10)

Province/State              Country       Lat        Long  \
ID                                                                            
0                            NaN          Afghanistan  33.93911   67.709953   
1                            NaN              Albania  41.15330   20.168300   
2                            NaN              Algeria  28.03390    1.659600   
3                            NaN              Andorra  42.50630    1.521800   
4                            NaN               Angola -11.20270   17.873900   
5                            NaN  Antigua and Barbuda  17.06080  -61.796400   
6                            NaN            Argentina -38.41610  -63.616700   
7                            NaN              Armenia  40.06910   45.038200   
8   Australian Capital Territory            Australia -35.47350  149.012400   
9                New South Wales            Australia -33.86880  151.209300   

          Date  Confirmed  Deaths  Recovered  Active                 Region  
ID                                                                           
0   2020-01-22          0       0          0       0  Eastern Mediterranean  
1   2020-01-22          0       0          0       0                 Europe  
2   2020-01-22          0       0          0       0                 Africa  
3   2020-01-22          0       0          0       0                 Europe  
4   2020-01-22          0       0          0       0                 Africa  
5   2020-01-22          0       0          0       0               Americas  
6   2020-01-22          0       0          0       0               Americas  
7   2020-01-22          0       0          0       0                 Europe  
8   2020-01-22          0       0          0       0        Western Pacific  
9   2020-01-22          0       0          0       0        Western Pacific

In [8]:
# Verificando se tem dados nulos na coluna Country/Region
dados_covid['Country'].isnull().sum(axis=0)


0

In [9]:
# Verificando se tem dados nulos na coluna Region
dados_covid['Region'].isnull().sum(axis=0)

0

## Primeiras Datas de Casos Confirmados

O primeiro caso confirmado do COVID-19 ocorreu em **Anhui na China** em **22 de Janeiro de 2020**

In [10]:
# Verificando a data do primeiros casos casos confirmados
primeiro_caso = dados_covid[(dados_covid['Confirmed'] > 0)]

# Exibindo os dados
primeiro_caso.head(10)

Province/State Country      Lat      Long        Date  Confirmed  Deaths  \
ID                                                                            
48          Anhui   China  31.8257  117.2264  2020-01-22          1       0   
49        Beijing   China  40.1824  116.4142  2020-01-22         14       0   
50      Chongqing   China  30.0572  107.8740  2020-01-22          6       0   
51         Fujian   China  26.0789  117.9874  2020-01-22          1       0   
53      Guangdong   China  23.3417  113.4244  2020-01-22         26       0   
54        Guangxi   China  23.8298  108.7881  2020-01-22          2       0   
55        Guizhou   China  26.8154  106.8748  2020-01-22          1       0   
56         Hainan   China  19.1959  109.7453  2020-01-22          4       0   
57          Hebei   China  39.5490  116.1306  2020-01-22          1       0   
59          Henan   China  37.8957  114.9042  2020-01-22          5       0   

    Recovered  Active           Region  
ID                                      
48          0       1  Western Pacific  
49          0      14  Western Pacific  
50          0       6  Western Pacific  
51          0       1  Western Pacific  
53          0      26  Western Pacific  
54          0       2  Western Pacific  
55          0       1  Western Pacific  
56          0       4  Western Pacific  
57          0       1  Western Pacific  
59          0       5  Western Pacific

O Primeiro caso com morte confirmado foi em **23 de Janeiro de 2020** na cidade de **Hebei na China**

In [11]:
# Exibindo a data com o primeiro caso com morte confirmada pelo COVID-19
dados_covid[(dados_covid['Deaths'] > 0) & (dados_covid['Confirmed'] < 10)]

Province/State         Country      Lat      Long        Date  \
ID                                                                             
318                     Hebei           China  39.5490  116.1306  2020-01-23   
579                     Hebei           China  39.5490  116.1306  2020-01-24   
580              Heilongjiang           China  47.8620  127.7615  2020-01-24   
840                     Hebei           China  39.5490  116.1306  2020-01-25   
841              Heilongjiang           China  47.8620  127.7615  2020-01-25   
...                       ...             ...      ...       ...         ...   
48009  British Virgin Islands  United Kingdom  18.4207  -64.6400  2020-07-23   
48270  British Virgin Islands  United Kingdom  18.4207  -64.6400  2020-07-24   
48531  British Virgin Islands  United Kingdom  18.4207  -64.6400  2020-07-25   
48792  British Virgin Islands  United Kingdom  18.4207  -64.6400  2020-07-26   
49053  British Virgin Islands  United Kingdom  18.4207  -64.6400  2020-07-27   

       Confirmed  Deaths  Recovered  Active           Region  
ID                                                            
318            1       1          0       0  Western Pacific  
579            2       1          0       1  Western Pacific  
580            4       1          0       3  Western Pacific  
840            8       1          0       7  Western Pacific  
841            9       1          0       8  Western Pacific  
...          ...     ...        ...     ...              ...  
48009          8       1          7       0           Europe  
48270          8       1          7       0           Europe  
48531          8       1          7       0           Europe  
48792          8       1          7       0           Europe  
49053          8       1          7       0           Europe  

[446 rows x 10 columns]

## Verificando o número de mortes por País

In [12]:
# Exibindo os paises
dados_covid['Country']


ID
0                  Afghanistan
1                      Albania
2                      Algeria
3                      Andorra
4                       Angola
                 ...          
49063    Sao Tome and Principe
49064                    Yemen
49065                  Comoros
49066               Tajikistan
49067                  Lesotho
Name: Country, Length: 49068, dtype: object

In [13]:
# Agrupando os paises
paises = dados_covid.groupby('Country')

# Exibindo os detalhes completos do dataframe
paises.describe()


Lat                                                 \
                    count       mean           std        min        25%   
Country                                                                    
Afghanistan         188.0  33.939110  7.124400e-15  33.939110  33.939110   
Albania             188.0  41.153300  0.000000e+00  41.153300  41.153300   
Algeria             188.0  28.033900  3.562200e-15  28.033900  28.033900   
Andorra             188.0  42.506300  0.000000e+00  42.506300  42.506300   
Angola              188.0 -11.202700  1.781100e-15 -11.202700 -11.202700   
...                   ...        ...           ...        ...        ...   
West Bank and Gaza  188.0  31.952200  0.000000e+00  31.952200  31.952200   
Western Sahara      188.0  24.215500  7.124400e-15  24.215500  24.215500   
Yemen               188.0  15.552727  1.781100e-15  15.552727  15.552727   
Zambia              188.0 -13.133897  0.000000e+00 -13.133897 -13.133897   
Zimbabwe            188.0 -19.015438  3.562200e-15 -19.015438 -19.015438   

                                                      Long             ...  \
                          50%        75%        max  count       mean  ...   
Country                                                                ...   
Afghanistan         33.939110  33.939110  33.939110  188.0  67.709953  ...   
Albania             41.153300  41.153300  41.153300  188.0  20.168300  ...   
Algeria             28.033900  28.033900  28.033900  188.0   1.659600  ...   
Andorra             42.506300  42.506300  42.506300  188.0   1.521800  ...   
Angola             -11.202700 -11.202700 -11.202700  188.0  17.873900  ...   
...                       ...        ...        ...    ...        ...  ...   
West Bank and Gaza  31.952200  31.952200  31.952200  188.0  35.233200  ...   
Western Sahara      24.215500  24.215500  24.215500  188.0 -12.885800  ...   
Yemen               15.552727  15.552727  15.552727  188.0  48.516388  ...   
Zambia             -13.133897 -13.133897 -13.133897  188.0  27.849332  ...   
Zimbabwe           -19.015438 -19.015438 -19.015438  188.0  29.154857  ...   

                   Recovered          Active                                 \
                         75%      max  count         mean          std  min   
Country                                                                       
Afghanistan          3091.25  25198.0  188.0  5792.829787  7059.917579  0.0   
Albania               985.25   2745.0  188.0   383.601064   513.101303  0.0   
Algeria              7119.25  18837.0  188.0  1839.819149  1873.709957  0.0   
Andorra               764.25    803.0  188.0   105.888298   156.003919  0.0   
Angola                 40.25    242.0  188.0    79.845745   146.445050  0.0   
...                      ...      ...    ...          ...          ...  ...   
West Bank and Gaza    410.00   3752.0  188.0   909.398936  1956.710157  0.0   
Western Sahara          6.00      8.0  188.0     1.010638     1.568368  0.0   
Yemen                  23.00    833.0  188.0   136.670213   179.730421  0.0   
Zambia                912.00   2815.0  188.0   229.611702   417.185481  0.0   
Zimbabwe               49.50    542.0  188.0   200.563830   387.999505  0.0   

                                                      
                      25%     50%       75%      max  
Country                                               
Afghanistan          4.00  1174.0  11683.50  20311.0  
Albania              1.50   205.5    331.75   1991.0  
Algeria             17.00  1331.0   2878.50   8114.0  
Andorra              1.00    23.5    158.00    504.0  
Angola               0.00    17.0     69.50    667.0  
...                   ...     ...       ...      ...  
West Bank and Gaza  18.25    75.0    257.25   7382.0  
Western Sahara       0.00     0.0      1.00      6.0  
Yemen                0.00     1.0    358.00    580.0  
Zambia               0.00    39.0    263.50   2236.0  
Zimbabwe             0.00    22.0    269.5

## Verificando o número de paises afetados pelo COVID-19

In [14]:
# Exibindo o número de paises afetados do dataframe
paises.count()

Province/State  Lat  Long  Date  Confirmed  Deaths  \
Country                                                                  
Afghanistan                      0  188   188   188        188     188   
Albania                          0  188   188   188        188     188   
Algeria                          0  188   188   188        188     188   
Andorra                          0  188   188   188        188     188   
Angola                           0  188   188   188        188     188   
...                            ...  ...   ...   ...        ...     ...   
West Bank and Gaza               0  188   188   188        188     188   
Western Sahara                   0  188   188   188        188     188   
Yemen                            0  188   188   188        188     188   
Zambia                           0  188   188   188        188     188   
Zimbabwe                         0  188   188   188        188     188   

                    Recovered  Active  Region  
Country                                        
Afghanistan               188     188     188  
Albania                   188     188     188  
Algeria                   188     188     188  
Andorra                   188     188     188  
Angola                    188     188     188  
...                       ...     ...     ...  
West Bank and Gaza        188     188     188  
Western Sahara            188     188     188  
Yemen                     188     188     188  
Zambia                    188     188     188  
Zimbabwe                  188     188     188  

[187 rows x 9 columns]

## Vamos Análisar o número de mortes

Nessa etapa vamos análisar o número de mortes no período analisado tanto o número total de mortes como também de casos confirmados e os paises afetados

In [15]:
# Total de Mortes
total_de_mortes = dados_covid['Deaths'].sum()

# Exibindo a série
total_de_mortes

43384903

In [16]:
# Total de mortes por país
mortes_por_paises = dados_covid.groupby('Country')['Deaths'].sum()

# Exibindo a série
mortes_por_paises.sort_values(ascending=False)

# Exportando o dataframe
mortes_por_paises.to_csv('total-de-mortes-por-paises.csv')

Com base nos Dados acima vimos que no perído de 187 dias o COVID-19 matou 43.384.903 de pessoas dos 187 paises analisados do dataframe

In [17]:
# Exibindo o país com o número de mortos
pais_com_maior_numero_de_mortes = dados_covid.sort_values('Deaths', ascending=False)

# Exibindo a série
pais_com_maior_numero_de_mortes

# Exportando os dados
pais_com_maior_numero_de_mortes.to_csv('paises-com-mais-mortes.csv')

In [18]:
# Agrupando por País
paises_unicos = dados_covid['Country'].unique()

# Agrupando os dados
paises_agrupados = dados_covid.loc[:,['Lat','Long','Deaths']]

# Exportando os dados para gerar o mapa
paises_agrupados.to_csv('paises.csv')


Com Base nos dados acima podemos conluir que o País com o maior número de mortes foi o Estados Unidos da América tanto no período analisado que foi de 187 dias com os incrivéis número de  11.011.411 mortos.

# Gerando Gráfico

Vamos ver os dados dos 5 paises com o maior número de mortes no gráfico com Mapa Folium

In [19]:
# Importando as bibliotecas
import numpy as np
import folium
from folium import plugins
from folium.plugins import HeatMap

# Gereando mapa em SVG

In [64]:
# Importando arquivo
dados_paises = np.genfromtxt('/content/drive/MyDrive/Colab Notebooks/Analise de Dados/Análise dos Dados do COVID-19/Total-Mortes.csv', delimiter=';')

# Criando o  mapa
mapa = folium.Map([0,0], zoom_start= 3, tiles="cartodbpositron")

# Exibindo o mapa
HeatMap(dados_paises).add_to(mapa)

mapa